app/services/attempts_api/attempt_event.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module AttemptsApi
+  class AttemptEvent
+    attr_reader :jti, :iat, :event_type, :session_id, :occurred_at, :event_metadata
+
+    def initialize(
+      event_type:,
+      session_id:,
+      occurred_at:,
+      event_metadata:,
+      jti: SecureRandom.uuid,
+      iat: Time.zone.now.to_i
+    )
+      @jti = jti
+      @iat = iat
+      @event_type = event_type
+      @session_id = session_id
+      @occurred_at = occurred_at
+      @event_metadata = event_metadata
+    end
+
+    def to_jwe(public_key:, issuer:)
+      jwk = JWT::JWK.new(public_key)
+
+      JWE.encrypt(
+        payload_json(issuer: issuer),
+        public_key,
+        typ: 'secevent+jwe',
+        zip: 'DEF',
+        alg: 'RSA-OAEP',
+        enc: 'A256GCM',
+        kid: jwk.kid,
+      )
+    end
+
+    def self.from_jwe(jwe, private_key)
+      decrypted_event = JWE.decrypt(jwe, private_key)
+      parsed_event = JSON.parse(decrypted_event)
+      event_type = parsed_event['events'].keys.first.split('/').last
```

Comment:
[question] is there ever a danger that the key/values are in a different order? (this feels a little brittle, but maybe i am being overly cautious)

Reply:
I think it's safe since `events` (currently) only ever contains one event. If there were multiple, we'd have to rethink it.

`.from_jwe` should also only ever be used in tests to reconstruct an instance of `AttemptEvent` from an event in Redis, so it should be ok.